lib/task-graph.js

```diff
@@ -220,25 +220,43 @@ function taskGraphFactory(
         // on failure (rare case probably, but still worth supporting IMO).
         definition.ignoreFailure = taskOverrides.ignoreFailure || false;
 
-        var allOptions = _.uniq(
-            _.keys(definition.options).concat(baseTaskDefinition.requiredOptions)
-        );
-
-        // If the graph has specifically defined options for a task, don't bother
-        // with whether they exist as a required option or not in the base definition.
-        if (_.has(self.definition.options, label)) {
-            allOptions = allOptions.concat(_.keys(self.definition.options[label]));
+        // If there is JSON schema defined for the task, then pass all task-specified option and
+        // defaults options to the task, let the schema to determine whether the additional options
+        // is allowed or not.
+        if (definition.hasOwnProperty('schemaRef')) {
```

Comment:
@benbp : This is the code that we had discussed before. The code is to pass all options that in `defaults` into all tasks. Previously we only pass those options only if that option in the option list of task definition. But now when the Task Schema is enabled, the `options` in the task definition doesn't reflect the whole supported options, because I have to remove those `null` or `[]` placeholder option as they violate the schema.

The will be a temporary workaround, I think the best solution is to find the whole supported option name from the schema. This is a challenge right now, as the schema may contain sub-schema. I plan to resolve this challenge later.